File: roleta/teste.py
import random
import time

arrayGeral = []
for i in range(0, 37):
    arrayGeral.append(i)

#print(arrayGeral)

pilha = [-1]
arraydiferenca = []
cont = 0
arrayMinutos = []
arraymedio = []
segundos=2
for j in range(1, 1000):
    for j in range(1, 2):
        numero = random.randint(0, 36)
        pilha.append(numero)
    print("pilha")
    print(pilha)
    arraydiferenca = list(filter(lambda x: (x not in pilha), arrayGeral))
    print("array diferenca")
    print(arraydiferenca)
    time.sleep(segundos)
    top5 = random.sample(arraydiferenca, 5)
    time.sleep(segundos)
    print("top 5")
    print(top5)
    time.sleep(segundos)
    if (len(top5)>=5):
        cont = 0
        numero = -1
        while (numero not in top5):
            cont += 1
            numero = random.randint(0, 36)
            print("numero sorteado")
            print(numero)
            time.sleep(segundos)
            print("cont")
            print(cont)
            time.sleep(segundos)
            #print(numero)
        arraymedio.append(cont)
        print("arraymedio")
        print(arraymedio)
    
    pilha = []


def Average(lst):
    return sum(lst) / len(lst)

#print(arraymedio)
print("maximo")
print(max(arraymedio))
print("media")
print(Average(arraymedio))
print("minimo")
print(min(arraymedio))
print("arraymedio")
print(arraymedio)

File: roleta/teste.js

const arrayGeral = []
let contador=0


function getRandomIntInclusive(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

  
function wait(ms){
    var start = new Date().getTime();
    var end = start;
    while(end < start + ms) {
        end = new Date().getTime();
    }
}

while (contador<36)
{
    contador++
    arrayGeral.push(contador)
}

let pilha = []
let arraydiferenca = []
let cont = 0
arrayMinutos = []

arrayTentativa=[]

cont=0
contGeral=0
while (contGeral<100)
{
    while (arraydiferenca.length != 4)
    {
        numero = getRandomIntInclusive(0, 36)
        pilha.push(numero)

        arraydiferenca = arrayGeral.filter(v => !pilha.includes(v))
        console.log(arraydiferenca)
        cont++  

        console.log(numero)
    }
    arrayTentativa.push(cont)
    contGeral++
    cont=0
    pilha=[]
    arraydiferenca=[]
}

const sum = arrayTentativa.reduce((a, b) => a + b, 0);
const avg = (sum / arrayTentativa.length) || 0;


console.log(arrayTentativa)
console.log("media"+avg)
